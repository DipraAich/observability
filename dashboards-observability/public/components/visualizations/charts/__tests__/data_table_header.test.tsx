import { waitFor } from '@testing-library/react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { GridHeader } from '../data_table/data_table_header';

const columns = [
  {
    lockVisible: true,
    columnsMenuParams: {
      suppressColumnFilter: true,
      suppressColumnSelectAll: true,
      suppressColumnExpandAll: true,
    },
    headerName: 'count()',
    field: 'count()',
    colId: 'count()',
  },
  {
    lockVisible: true,
    columnsMenuParams: {
      suppressColumnFilter: true,
      suppressColumnSelectAll: true,
      suppressColumnExpandAll: true,
    },
    headerName: 'span(timestamp,1d)',
    field: 'span(timestamp,1d)',
    colId: 'span(timestamp,1d)',
  },
];

describe('Test data table header component', () => {
  configure({ adapter: new Adapter() });

  it('Renders data table header component', async () => {
    const wrapper = mount(
      <GridHeader
        isFullScreen={false}
        setIsFullScreenHandler={() => {}}
        selectedRowDensity={{
          icon: 'tableDensityCompact',
          height: 35,
          selected: true,
        }}
        selectDensityHandler={() => {}}
        columnVisiblityHandler={() => {}}
        columns={columns}
        columnVisibility={[]}
      />
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
