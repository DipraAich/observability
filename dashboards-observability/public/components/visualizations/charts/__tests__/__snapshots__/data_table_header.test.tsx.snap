// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test data table header component Renders data table header component 1`] = `
<GridHeader
  columnVisibility={Array []}
  columnVisiblityHandler={[Function]}
  columns={
    Array [
      Object {
        "colId": "count()",
        "columnsMenuParams": Object {
          "suppressColumnExpandAll": true,
          "suppressColumnFilter": true,
          "suppressColumnSelectAll": true,
        },
        "field": "count()",
        "headerName": "count()",
        "lockVisible": true,
      },
      Object {
        "colId": "span(timestamp,1d)",
        "columnsMenuParams": Object {
          "suppressColumnExpandAll": true,
          "suppressColumnFilter": true,
          "suppressColumnSelectAll": true,
        },
        "field": "span(timestamp,1d)",
        "headerName": "span(timestamp,1d)",
        "lockVisible": true,
      },
    ]
  }
  isFullScreen={false}
  selectDensityHandler={[Function]}
  selectedRowDensity={
    Object {
      "height": 35,
      "icon": "tableDensityCompact",
      "selected": true,
    }
  }
  setIsFullScreenHandler={[Function]}
>
  <EuiFlexGroup
    gutterSize="none"
    justifyContent="flexStart"
    responsive={true}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="euiFlexGroup euiFlexGroup--directionRow euiFlexGroup--responsive"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <EuiFlexItem
        style={
          Object {
            "maxWidth": "150px",
          }
        }
      >
        <div
          className="euiFlexItem"
          style={
            Object {
              "maxWidth": "150px",
            }
          }
        >
          <DensityPopover
            selectDensityHandler={[Function]}
            selectedDensity={
              Object {
                "height": 35,
                "icon": "tableDensityCompact",
                "selected": true,
              }
            }
          >
            <EuiPopover
              anchorPosition="downCenter"
              button={
                <EuiButtonEmpty
                  aria-label="Next"
                  color="text"
                  iconSize="s"
                  iconType="tableDensityCompact"
                  onClick={[Function]}
                >
                  Density
                </EuiButtonEmpty>
              }
              closePopover={[Function]}
              display="inlineBlock"
              hasArrow={true}
              isOpen={false}
              ownFocus={true}
              panelPaddingSize="m"
            >
              <div
                className="euiPopover euiPopover--anchorDownCenter"
              >
                <div
                  className="euiPopover__anchor"
                >
                  <EuiButtonEmpty
                    aria-label="Next"
                    color="text"
                    iconSize="s"
                    iconType="tableDensityCompact"
                    onClick={[Function]}
                  >
                    <button
                      aria-label="Next"
                      className="euiButtonEmpty euiButtonEmpty--text"
                      disabled={false}
                      iconSize="s"
                      onClick={[Function]}
                      type="button"
                    >
                      <EuiButtonContent
                        className="euiButtonEmpty__content"
                        iconSide="left"
                        iconSize="m"
                        iconType="tableDensityCompact"
                        textProps={
                          Object {
                            "className": "euiButtonEmpty__text",
                          }
                        }
                      >
                        <span
                          className="euiButtonContent euiButtonEmpty__content"
                        >
                          <EuiIcon
                            className="euiButtonContent__icon"
                            color="inherit"
                            size="m"
                            type="tableDensityCompact"
                          >
                            <EuiIconEmpty
                              aria-hidden={true}
                              className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                              focusable="false"
                              role="img"
                              style={null}
                            >
                              <svg
                                aria-hidden={true}
                                className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                                focusable="false"
                                height={16}
                                role="img"
                                style={null}
                                viewBox="0 0 16 16"
                                width={16}
                                xmlns="http://www.w3.org/2000/svg"
                              />
                            </EuiIconEmpty>
                          </EuiIcon>
                          <span
                            className="euiButtonEmpty__text"
                          >
                            Density
                          </span>
                        </span>
                      </EuiButtonContent>
                    </button>
                  </EuiButtonEmpty>
                </div>
              </div>
            </EuiPopover>
          </DensityPopover>
        </div>
      </EuiFlexItem>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": "150px",
          }
        }
      >
        <div
          className="euiFlexItem"
          style={
            Object {
              "maxWidth": "150px",
            }
          }
        >
          <ColumnVisiblityPopover
            columnVisibility={Array []}
            columnVisiblityHandler={[Function]}
            columns={
              Array [
                Object {
                  "colId": "count()",
                  "columnsMenuParams": Object {
                    "suppressColumnExpandAll": true,
                    "suppressColumnFilter": true,
                    "suppressColumnSelectAll": true,
                  },
                  "field": "count()",
                  "headerName": "count()",
                  "lockVisible": true,
                },
                Object {
                  "colId": "span(timestamp,1d)",
                  "columnsMenuParams": Object {
                    "suppressColumnExpandAll": true,
                    "suppressColumnFilter": true,
                    "suppressColumnSelectAll": true,
                  },
                  "field": "span(timestamp,1d)",
                  "headerName": "span(timestamp,1d)",
                  "lockVisible": true,
                },
              ]
            }
          >
            <EuiPopover
              anchorPosition="downCenter"
              button={
                <EuiButtonEmpty
                  aria-label="Next"
                  color="text"
                  iconSize="s"
                  iconType="listAdd"
                  onClick={[Function]}
                >
                  Columns
                </EuiButtonEmpty>
              }
              closePopover={[Function]}
              display="inlineBlock"
              hasArrow={true}
              isOpen={false}
              ownFocus={true}
              panelPaddingSize="m"
            >
              <div
                className="euiPopover euiPopover--anchorDownCenter"
              >
                <div
                  className="euiPopover__anchor"
                >
                  <EuiButtonEmpty
                    aria-label="Next"
                    color="text"
                    iconSize="s"
                    iconType="listAdd"
                    onClick={[Function]}
                  >
                    <button
                      aria-label="Next"
                      className="euiButtonEmpty euiButtonEmpty--text"
                      disabled={false}
                      iconSize="s"
                      onClick={[Function]}
                      type="button"
                    >
                      <EuiButtonContent
                        className="euiButtonEmpty__content"
                        iconSide="left"
                        iconSize="m"
                        iconType="listAdd"
                        textProps={
                          Object {
                            "className": "euiButtonEmpty__text",
                          }
                        }
                      >
                        <span
                          className="euiButtonContent euiButtonEmpty__content"
                        >
                          <EuiIcon
                            className="euiButtonContent__icon"
                            color="inherit"
                            size="m"
                            type="listAdd"
                          >
                            <EuiIconEmpty
                              aria-hidden={true}
                              className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                              focusable="false"
                              role="img"
                              style={null}
                            >
                              <svg
                                aria-hidden={true}
                                className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                                focusable="false"
                                height={16}
                                role="img"
                                style={null}
                                viewBox="0 0 16 16"
                                width={16}
                                xmlns="http://www.w3.org/2000/svg"
                              />
                            </EuiIconEmpty>
                          </EuiIcon>
                          <span
                            className="euiButtonEmpty__text"
                          >
                            Columns
                          </span>
                        </span>
                      </EuiButtonContent>
                    </button>
                  </EuiButtonEmpty>
                </div>
              </div>
            </EuiPopover>
          </ColumnVisiblityPopover>
        </div>
      </EuiFlexItem>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": "150px",
          }
        }
      >
        <div
          className="euiFlexItem"
          style={
            Object {
              "maxWidth": "150px",
            }
          }
        >
          <EuiButtonEmpty
            aria-label="Next"
            color="text"
            iconSize="s"
            iconType="fullScreen"
            onClick={[Function]}
          >
            <button
              aria-label="Next"
              className="euiButtonEmpty euiButtonEmpty--text"
              disabled={false}
              iconSize="s"
              onClick={[Function]}
              type="button"
            >
              <EuiButtonContent
                className="euiButtonEmpty__content"
                iconSide="left"
                iconSize="m"
                iconType="fullScreen"
                textProps={
                  Object {
                    "className": "euiButtonEmpty__text",
                  }
                }
              >
                <span
                  className="euiButtonContent euiButtonEmpty__content"
                >
                  <EuiIcon
                    className="euiButtonContent__icon"
                    color="inherit"
                    size="m"
                    type="fullScreen"
                  >
                    <EuiIconEmpty
                      aria-hidden={true}
                      className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                      focusable="false"
                      role="img"
                      style={null}
                    >
                      <svg
                        aria-hidden={true}
                        className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                        focusable="false"
                        height={16}
                        role="img"
                        style={null}
                        viewBox="0 0 16 16"
                        width={16}
                        xmlns="http://www.w3.org/2000/svg"
                      />
                    </EuiIconEmpty>
                  </EuiIcon>
                  <span
                    className="euiButtonEmpty__text"
                  >
                    Full screen
                  </span>
                </span>
              </EuiButtonContent>
            </button>
          </EuiButtonEmpty>
        </div>
      </EuiFlexItem>
    </div>
  </EuiFlexGroup>
</GridHeader>
`;
